paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/clone/CopyFileOperator.java

```diff
@@ -139,6 +141,37 @@ private Path pathOfTable(Table table) {
         return new Path(table.options().get(CoreOptions.PATH.key()));
     }
 
+    private void copyManifestFile(Path sourcePath, Path targetPath, CloneFileInfo cloneFileInfo)
```

Comment:
Why copy ManifestFile by this new method?

Reply:
Because source table's manifest's file entry have `DataFileMeta` field,  in this commit[1], we have add `externalPath` in `DataFileMeta`, so when we clone source table's manifest, we should rewrite the manifest file, rewrite the `externalPath` in `DataMileMeta` to null, (being null means that the datafile exists in the warehouse path).
[1] https://github.com/apache/paimon/commit/9ff5192a09856aff318597e589f8e8c8f6ff325b

Reply:
I think in clone action, maybe it is better to don't change file's content.

Reply:
However, since we introduced the `externalPath`, which points to the data storage location of the source table, we need to modify this property to the `warehouse path` of the target table when cloning.
Do you have any good suggestions to solve this problem?